### PR TITLE
Add counts

### PR DESCRIPTION
**Pull request by matt-malone**
Added 2 properties:

1. viewCount - tracks the number of times the component enters the viewport
2. leaveCount - tracks the number of times the component leaves the viewport

I found these very useful when applying a "fade" effect that I only wanted to happen the first time the component scrolled into view.

### CODE REVIEW

**Comment by roderickhsiao** [comment]
Looks good to me, I will update the storybook https://roderickhsiao.github.io/react-in-viewport/ to reflect the change

**Comment by roderickhsiao** [comment]
@matt-malone Having a second thought for this, do you think it makes sense for parent container to calculate the count via `onEnterViewport` and `onLeaveViewport`. In your example, we could use `MySectionBlock` internal state to handle the loading state. How do you think?

**Comment by roderickhsiao** [comment]
In your case you could also use `disconnectOnLeave` so that only this first time will have the props change

**Comment by matt-malone** [comment]
Hmm, I did initially try that but the prop never got updated. Let me try again. I had some issues in chrome with the onEnterViewport/onLeaveViewport handlers that might have caused the props to not update. Now that I fixed that, maybe tracking in the parent will work. I'll report back :)

**Comment by matt-malone** [comment]
Ahh yes, I remember now. Couple of issues when doing this from the parent:

1. Tracking viewCount in the parent is clunky when I have multiple elements (for instance I have about 5 sections I want to fade in as they enter the viewport). I have to ref all of them and orchestrate them from the parent. Whereas putting them in the viewport component allows me to encapsulate the logic my custom component.
2. Lastly I am having trouble getting the handler to re-render the children when it is triggered. So even if I do increment in the handler, the change is not reflected in the children. Basically I would expect this to work:
```javascript
myViewCount={this.state.viewCount}
onEnterViewport={()=>this.setState({viewCount:this.state.viewCount + 1}) }
```
but it hasn't worked for me so far whereas adding it to the InViewport component has worked as expected.

`disconnectOnLeave` might work though it is not exactly the behavior I'm looking for, I'd need to restructure a bit to take advantage of it since I'm actually applying styles on several different counts. If the pull request isn't approved I'll just have to weigh restructuring vs using a non-standard version of in viewport. :)

**Comment by roderickhsiao** [comment]
Thanks, its fine to include in the library, just want to see if this is the common use case :)

Thanks a lot for going into details

**Comment by roderickhsiao** [comment]
`0.0.29`. I update slightly the naming to use `enterCount` and `leaveCount` to better match the interface
